third_party/txt/src/txt/paragraph.cc

```diff
@@ -549,6 +549,19 @@ void Paragraph::Layout(double width, bool force) {
             ? line_range.end_excluding_whitespace
             : line_range.end;
 
+    // Build a map of styled runs indexed by start position to TextDirection.
+    // Once the direction of a non-ghost line run has been established for the
+    // styled run, we will enforce that the ghost run use the direction of the
+    // most recently seen bidi run that is part of the styled run. This is to
+    // ensure that default-LTR whitespace gets the proper directionality when
+    // used in a RTL run.
+    std::map<size_t, TextDirection> styled_run_direction_map;
+    for (size_t i = 0; i < runs_.size(); ++i) {
+      StyledRuns::Run run = runs_.GetRun(i);
+      // Initialize TextDirection to LTR as that is what ICU initializes it to.
+      styled_run_direction_map.emplace(
```

Comment:
I don't think the emplace adds much value here as the enum value is not expensive to copy.